androidApp/src/main/java/co/nimblehq/avishek/phong/kmmic/android/ui/screen/home/HomeScreen.kt

```diff
@@ -1,21 +1,229 @@
+        appVersion = "v0.5.0"
+    }
+
+    HomeContentWithDrawer(
+        appVersion = appVersion,
+        scaffoldState = scaffoldState,
+        currentDate = currentDate,
+        isLoading = isLoading,
+        surveyUiModels = surveyUiModels,
+        userUiModel = userUiModel
+    )
+}
+
+@Composable
+private fun HomeContentWithDrawer(
+    appVersion: String,
+    initialDrawerState: DrawerValue = DrawerValue.Closed,
+    scaffoldState: ScaffoldState,
+    currentDate: String,
+    userUiModel: UserUiModel? = null,
+    surveyUiModels: List<SurveyUiModel>,
+    isLoading: Boolean,
+) {
+    val drawerState = rememberDrawerState(initialDrawerState)
+    val scope = rememberCoroutineScope()
+
+    RtlModalDrawer(
+        drawerState = drawerState,
+        gesturesEnabled = drawerState.isOpen,
+        drawerContent = {
+            HomeDrawer(
+                userUiModel = userUiModel,
+                appVersion = appVersion,
+                onLogoutClick = {
+                    scope.launch { drawerState.close() }
+                }
+            )
+        }
+    ) {
+        HomeContent(
+            scaffoldState = scaffoldState,
+            userUiModel = userUiModel,
+            currentDate = currentDate,
+            surveyUiModels = surveyUiModels,
+            isLoading = isLoading,
+            onUserAvatarClick = {
+                scope.launch { drawerState.open() }
+            }
+        )
+    }
 }
 
+@OptIn(ExperimentalFoundationApi::class)
 @Composable
-fun HomeContent(modifier: Modifier = Modifier) {
-    Text(
-        text = "Home under construction",
-        style = MaterialTheme.typography.h1,
+fun HomeContent(
+    scaffoldState: ScaffoldState,
+    userUiModel: UserUiModel?,
+    isLoading: Boolean,
+    currentDate: String,
+    surveyUiModels: List<SurveyUiModel>,
+    onUserAvatarClick: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    val pagerState = rememberPagerState()
+    var surveyUiModel by remember { mutableStateOf<SurveyUiModel?>(null) }
+
+    LaunchedEffect(surveyUiModels) {
+        snapshotFlow { pagerState.currentPage }.collect { page ->
+            surveyUiModel = surveyUiModels.getOrNull(page)
+        }
+    }
+
+    Scaffold(
+        scaffoldState = scaffoldState,
         modifier = modifier
-    )
+    ) { padding ->
+        Box(
+            modifier = Modifier
+                .fillMaxSize()
+                .padding(padding)
+        ) {
+            HorizontalPager(
+                pageCount = surveyUiModels.size,
+                state = pagerState,
+                modifier = Modifier
+                    .fillMaxSize()
+                    .background(color = BlackRussian)
+            ) { page ->
+                Box(
+                    Modifier
+                        .pagerFadeTransition(page, pagerState)
+                        .fillMaxSize()
+                ) {
+                    Box(modifier = Modifier.fillMaxSize()) {
```

Comment:
Duplicated `Box`. Let's apply `pagerFadeTransition` to the second `Box` and remove the first one.